lib/test/KettleTestUtils.http.js

```diff
@@ -51,9 +54,86 @@ fluid.defaults("kettle.test.request", {
     termMap: {}
 });
 
+// A component that sends multipart/form-data via POST
+// using the form-data module
+// Can be used to send both files and fields
+// Used currently for testing the multer-based middleware
+//
+// TODO: Should elements of this and kettle.test.request.http be merged?
+// How to best do this while maintaing backwards compatibility?
+fluid.defaults("kettle.test.request.formData", {
+    gradeNames: ["kettle.test.request"],
+    events: { // this will fire with the signature (data, that)
+        onComplete: null
+    },
+    invokers: {
+        send: {
+            funcName: "kettle.test.request.formData.send",
+            args: ["{that}", "{that}.events.onComplete.fire"]
+        }
+    },
+    formData: {
+        // key-value structure where key will be the appended field
+        // name and the value is one of
+        // 1. the path to a file to be attached
+        // 2. an array of file paths
+        files: {
+            // "file": "./LICENSE.txt"
+            // "files": "./LICENSE.txt, ./README.md"
+        },
+        fields: {
+            // key-value structure where key will be the appended field
+            // name and the value is the string value to be associated
+        }
+    }
+});
 
+kettle.test.request.formData.send = function (that, callback) {
 
+    var requestOptions = that.options;
 
+    var submitURL = "http://" + requestOptions.hostname + ":" + requestOptions.port + requestOptions.path;
+
+    var form = new FormData();
+
+    // Append all files
+    fluid.each(requestOptions.formData.files, function (filePath, fieldKey) {
+        // Single file path
+        if (typeof filePath === "string") {
```

Comment:
Easier is
var filePaths = fluid.makeArray(filePath);
filePaths.forEach(function (filePath) {
    var resolved = fluid.module.resolvePath(filePath);
    form.append(fieldKey, fs.createReadStream(resolved));
});